halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/providers/google/GoogleAccount.java

```diff
@@ -47,14 +47,15 @@
   @LocalFile String userDataFile;
   private List<String> regions;
 
-  @Autowired
-  private SecretSessionManager secretSessionManager;
-
   @JsonIgnore
-  public GoogleNamedAccountCredentials getNamedAccountCredentials(String version, ConfigProblemSetBuilder p) {
+  public GoogleNamedAccountCredentials getNamedAccountCredentials(String version, SecretSessionManager secretSessionManager, ConfigProblemSetBuilder p) {
     String jsonKey = null;
     if (!StringUtils.isEmpty(getJsonPath())) {
-      jsonKey = secretSessionManager.validatingFileDecrypt(p, getJsonPath());
+      if (secretSessionManager != null) {
```

Comment:
In what cases will this case occur?

Reply:
This method is called from two places:
* `GoogleAccountValidator` - this class has a functioning `SecretSessionManager` so it won't be null
* `GoogleProviderUtils` - doesn't have `SecretSessionManager` and secret decryption isn't supported in this flow, so we're passing in null. The credentials will be read in the way they were before, using `ValidatingFileReader.contents(p, getJsonPath())`

Reply:
Sounds good.